onnx/shape_inference/implementation.cc

```diff
@@ -214,6 +214,7 @@ static void InferShapesImpl(
 	}
 
     try {
+      schema->Verify(n, &valueTypesByName);
```

Comment:
Can we separate out the basic-verification done by Verify previously from the new checks you added into separate functions? I believe that the idea was that in TypeAndShapeInferenceFunction one could assume that the basic-verification has happened successfully … so it doesn't need to check if a required-attribute is missing, etc. The new checks you added looks like it has to happen after the TypeAndShapeInferenceFunction is called. So, we may need to split it.

Reply:
Good idea - like this?

Reply:
Yes, looks good, thanks! 

Can we also extend this checker:https://github.com/onnx/onnx/blob/c963586d0f8dd5740777b2fd06f04ec60816de9f/onnx/checker.py#L87 

Extend it to do the basic checking (as currently) and then call shape-inference for the type-inference and checking? May be it can be optionally invoked (controlled by an optional parameter) so that we preserve the existing behavior if invoked as currently?

Reply:
CR updated according with new checking routine, let's see.